tensorflow_toolkit/nlp/bert/README.md

```diff
@@ -0,0 +1,206 @@
+# Fine tuning Bert base XNLI and Squad 1.1 from Google Bert
+
+## Prerequistes
+
+	*Ubuntu 18.02
+	*Tensorflow 1.1X
+	*OpenVINO 2020.1
+
+## TensorFlow BERT Installation
+
+1. Clone the google bert
+
+	```bash
+	$ git clone https://github.com/google-research/bert
+	$ cd bert
+	```
+2. This folder contains a git patch to add support to export the saved model from the trained model for xnli and squad.
+   Apply the `0001-Add-support-to-export-the-saved-model-to-the-run_cla.patch` file to the google bert repository
+
+	```bash
+	$ git checkout eedf5716ce1268e56f0a50264a88cafad334ac61
+	$ git am 0001-Add-support-to-export-the-saved-model-to-the-run_cla.patch
+	```
+
+### Fine Tuning BERT-Base with XNLI dataset
+
+1. Download the pre-trained Bert-base chinese model
+
+	```bash
+	$ wget https://storage.googleapis.com/bert_models/2018_11_03/chinese_L-12_H-768_A-12.zip
+	```
+2. Refer the Fine-tuning Example section of the multilingual google wiki for the data set and finetuning
+
+	```bash
+	https://github.com/google-research/bert/blob/master/multilingual.md
+	```
+	```bash
+	$ export BERT_BASE_DIR=/path/to/bert/chinese_L-12_H-768_A-12
+	$ export XNLI_DIR=/path/to/xnli
+	$ pip install tensorflow==1.14
+	```
+	```bash
+	$ python run_classifier.py \
+	--task_name=XNLI \
+	--do_train=true \
+	--do_eval=true \
+	--data_dir=$XNLI_DIR \
+	--vocab_file=$BERT_BASE_DIR/vocab.txt \
+	--bert_config_file=$BERT_BASE_DIR/bert_config.json \
+	--init_checkpoint=$BERT_BASE_DIR/bert_model.ckpt \
+	--max_seq_length=128 \
+	--train_batch_size=32 \
+	--learning_rate=5e-5 \
+	--num_train_epochs=2.0 \
+	--output_dir=/tmp/xnli_output/
+	```
+	With the Chinese-only model, the result:
+
+	```bash
+	eval_accuracy = 0.774116
+	eval_loss = 0.83554
+	global_step = 24543
+	loss = 0.74603
+	```
+3. Export the saved model from the trained checkpoint
+
+	```bash
+	$python run_classifier.py \
+	--task_name=XNLI \
+	--do_eval=true \
+	--data_dir=$XNLI_DI \
+	--vocab_file=$BERT_BASE_DIR/vocab.txt \
+	--bert_config_file=$BERT_BASE_DIR/bert_config.json \
+	--init_checkpoint=$BERT_BASE_DIR/model.ckpt-24543 \
+	--max_seq_length=128 \
+	--train_batch_size=32 \
+	--learning_rate=5e-5 \
+	--num_train_epochs=2.0 \
+	--output_dir=/tmp/xnli_output \
+	--export_dir=/tmp/xnli_output/export_model
+	```
+4. Clone the tensorflow source and generate the frozen graph
+
+	```bash
+	$ git clone -b r1.14 --single-branch https://github.com/tensorflow/tensorflow.git
+	$ cd tensorflow
+	$ python tensorflow/python/tools/freeze_graph.py \
+	--input_saved_model_dir=/tmp/xnli_output/export_model/<folder>/ \
+	--output_graph= ./bert_xnli_fp32_graph.pb \
+	--output_node_names=loss/LogSoftmax
+	```
+5. Run the OpenVino Model Optimizer to generate the IR
+
+	```bash
+	$ python $MO_DIR/mo.py --framework=tf \
```

Comment:
Write as 
```
mo.py --framework=tf \
...
```
Directory with `mo.py` is added  to `PATH` by  `source /opt/intel/openvino/bin/setupvars.sh`.

Reply:
ok